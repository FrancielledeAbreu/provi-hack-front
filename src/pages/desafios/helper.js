export const Forum = {
  name: "Fórum",
  content: [
    ". O que é HackaWork",
    ". Empresas parceiras",
    ". Como se portar na entrevista?",
    ". Dicas valiosas",
  ],
};

export const Eventos = {
  name: "Eventos",
  content: [
    "7/fev - 10h: Bate papo Previ",
    "8/fev - 10h: Live com Wavy Global",
    "9/fev - 10h: Live com Única",
    "10/fev - 10h: Live com Pagar.me",
  ],
};
