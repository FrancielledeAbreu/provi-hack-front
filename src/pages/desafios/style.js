import styled from "styled-components";
import { Button } from "antd";

export const Header = styled.div`
  @import url("https://fonts.googleapis.com/css2?family=Roboto:wght@100;500&display=swap");
  font-family: "Roboto", sans-serif;
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding-left: 15%;
  padding-right: 15%;
  padding-top: 5%;
  padding-bottom: 5%;
  align-items: center;

  span {
    margin-right: 10px;
  }
`;

export const NewButton = styled(Button)`
  margin-left: 10px;
`;
export const Nav = styled.div`
  width: 80%;
  display: flex;
  justify-content: space-around;
  padding-left: 10%;
  padding-right: 10%;
  background-color: #c4c4c4;
  padding: 1%;
  margin-left: 10%;
`;
