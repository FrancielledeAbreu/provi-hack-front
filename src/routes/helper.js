import Home from "../pages/home";
import Login from "../pages/login";
import Desafios from "../pages/desafios";
import ChallengeDetails from "../pages/challengeDetails";

export const routes = [
  { path: "/", name: "home", component: Home },
  { path: "/login", name: "login", component: Login },
  { path: "/desafios", name: "desafios", component: Desafios },
  {
    path: "/challenge-details/:id",
    name: "ChallengeDetails",
    component: ChallengeDetails,
  },
];
