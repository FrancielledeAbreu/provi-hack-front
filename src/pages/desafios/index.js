import React, { useEffect } from "react";
import { UserOutlined } from "@ant-design/icons";
import { motion } from "framer-motion";
import { Input, Button, Avatar } from "antd";
import { useDispatch, useSelector } from "react-redux";

//locals
import PopUp from "../../components/popUp";
import CardChallenge from "../../components/challenge";
import { Forum, Eventos } from "./helper";
import {
  challengesRequest,
  challengesRequestTeam,
  challengesRequestSingle,
  challengesRequestFilterName,
} from "../../redux/actions/challenges";

//style
import {
  Header,
  Nav,
  NewButton,
  Challenges,
  Message,
  ContainerPopup,
} from "./style";

const Desafios = () => {
  const dispatch = useDispatch();
  const challenges = useSelector((state) => state.challengeReducer);
  const { Search } = Input;

  useEffect(() => {
    dispatch(challengesRequest());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleChallengesAll = () => {
    dispatch(challengesRequest());
  };

  const handleFilterTeam = () => {
    dispatch(challengesRequestTeam());
  };

  const handleFilterSingle = () => {
    dispatch(challengesRequestSingle());
  };

  const onSearch = (value) => {
    dispatch(challengesRequestFilterName(value));
  };

  return (
    <div data-testid="Challenges">
      <Header>
        <NewButton>sou uma empresa parceira</NewButton>
        <div>
          <span>Meus desafios</span>
          <Avatar size={64} icon={<UserOutlined />} />
        </div>
      </Header>
      <Nav>
        <Button onClick={handleFilterTeam} type="primary">
          Desafio em grupo
        </Button>
        <Button onClick={handleFilterSingle} type="primary">
          Desafio individual
        </Button>
        <Search
          placeholder="buscar um desafio"
          onSearch={onSearch}
          enterButton
          style={{ width: "30%" }}
        />
      </Nav>
      <Challenges>
        <div>
          {challenges.length > 0 ? (
            challenges.map(({ name, challenges_type, image, level }, index) => (
              <motion.div
                whileHover={{ scale: 1.1 }}
                whileTap={{ scale: 0.8 }}
                style={{ padding: "2%", marginLeft: "8%" }}
              >
                <CardChallenge
                  name={name}
                  challenges_type={challenges_type}
                  image={image}
                  level={level}
                />
              </motion.div>
            ))
          ) : (
            <Message>
              <div>NÃ£o existe desafio para este filtro</div>
              <Button onClick={handleChallengesAll} type="primary">
                Todos Desafios
              </Button>
            </Message>
          )}
        </div>
        <ContainerPopup>
          <PopUp object={Forum} />
          <PopUp object={Eventos} />
        </ContainerPopup>
      </Challenges>
    </div>
  );
};

export default Desafios;
