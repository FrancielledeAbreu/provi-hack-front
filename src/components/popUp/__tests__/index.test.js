import React from "react";
import PopUp from "../index";
import renderer from "react-test-renderer";
import { Forum } from "../../../pages/desafios/helper";

describe("PopUp Test", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PopUp object={Forum} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
