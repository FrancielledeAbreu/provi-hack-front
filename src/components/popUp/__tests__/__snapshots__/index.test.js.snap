// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp Test renders correctly 1`] = `
<div
  className="sc-bdfBwQ fBhDya"
>
  <h1>
    Fórum
  </h1>
  <div
    style={Object {}}
  >
    <div
      className="sc-gsTCUz dLNejD"
    >
      . O que é HackaWork
    </div>
  </div>
  <div
    style={Object {}}
  >
    <div
      className="sc-gsTCUz dLNejD"
    >
      . Empresas parceiras
    </div>
  </div>
  <div
    style={Object {}}
  >
    <div
      className="sc-gsTCUz dLNejD"
    >
      . Como se portar na entrevista?
    </div>
  </div>
  <div
    style={Object {}}
  >
    <div
      className="sc-gsTCUz dLNejD"
    >
      . Dicas valiosas
    </div>
  </div>
</div>
`;
